### PR TITLE
Add k8s examples - Tanka & Plain YAML

### PR DESCRIPTION
**Pull request by Temikus**
Additionally, fix docker command examples in readme

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 bolt.db
 metrics.yaml
 Vagrantfile
+k8s/**/*
```
Comment (rjerrems) [inline .dockerignore:6]: nit: newline

File: k8s/tanka/environments/default/config/metrics.jsonnet.example [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  "stackdriver_destinations": [
+    {
+      "name": "stackdriver"
+    },
+    {
+      "name": "another_stackdriver",
+      "project_id": "another-projectname"
+    }
+  ]
+}
```
Comment (rjerrems) [inline k8s/tanka/environments/default/config/metrics.jsonnet.example]: nit: newline

File: k8s/yaml/persistent_volume_claim.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+kind: PersistentVolumeClaim
+metadata:
+  name: ts-bridge-pv-claim
+spec:
+  storageClassName: ssd-backed
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 100Gi
```
Comment (rjerrems) [inline k8s/yaml/persistent_volume_claim.yaml]: nit: newline

File: k8s/yaml/secret.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: google-api-credentials
+type: Opaque
+data:
+  #json_key: <contents of running 'base64 the-downloaded-SA-credentials.json'>
+  json_key: REPLACE_ME
```
Comment (rjerrems) [inline k8s/yaml/secret.yaml]: nit: newline

File: k8s/yaml/service.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  name: ts-bridge-service
+spec:
+  type: ClusterIP
+  selector:
+    app: ts-bridge
+  ports:
+    - protocol: TCP
+      port: 8080
+      targetPort: 8080
```
Comment (rjerrems) [inline k8s/yaml/service.yaml]: nit: newline

File: k8s/yaml/storage_class.yaml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+---
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: ssd-backed
+provisioner: kubernetes.io/gce-pd
+parameters:
+  type: pd-ssd
```
Comment (rjerrems) [inline k8s/yaml/storage_class.yaml]: nit: newline

File: k8s/yaml/README.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+This provides an extremely basic "bare-bones" GKE deployment example for K8s using:
+ - BoltDB and PD-SSD for persistence
+ - Basic `ClusterIP` service for exposing the web ui
+
+## Quickstart
+
+1) Edit the configuration in `config_map.yaml`
+
+2) Add the service account key secret in secret.yaml by parsing an account key using `base64`, e.g.:
```
Comment (rjerrems) [inline k8s/yaml/README.md:11]: It may be worthwhile to have a brief comment that this is only required if not using workload identity in GKE.

**Review by rjerrems — COMMENTED** [review]
Overall its looking good, I have added a few cosmetic comments but should be good to go after.

File: k8s/tanka/README.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    ```
+
+3) You can set the endpoint from your current context by issuing this command:
+
+    ```
+    $ SERVER_ENDPOINT=$(kubectl config view -o jsonpath='{"Cluster name\tServer\n"}{range .clusters[*]}{.name}{"\t"}{.cluster.server}{"\n"}{end}' | grep `kubectl config current-context` | cut -f2)
+    $ tk env set environments/default --server=${SERVER_ENDPOINT}
+    ```
+
+4) Configure the deployment:P
```
[the remark this answers is not in the document]
Comment (Temikus) [reply k8s/tanka/README.md:29]: No, typo.

File: k8s/tanka/README.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+    Edit the `environments/default/main.jsonnet.example` to specify running parameters and move it to live config:
+    
+    ```
+    $ mv environments/default/main.jsonnet.example environments/default/main.jsonnet
+    ```
+    Note: you can convert an existing YAML config with `yq` utility, e.g.:
```
[the remark this answers is not in the document]
Comment (Temikus) [reply k8s/tanka/README.md:36]: Done with yq.

Config is included actually, just README flows badly. Fixed.

File: k8s/tanka/environments/default/main.jsonnet.example [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+      // Binary arguments
+      args: [
+        '--storage-engine=boltdb',
+        '--enable-status-page',
+        '--stats-sd-project=REPLACE_ME',
+        '--update-parallelism=4',
+        '--sync-period=10s',
+      ],
+      auth: {
+        // Path to JSON Service account key which will be automatically parsed into an opaque k8s secret
```
[the remark this answers is not in the document]
Comment (Temikus) [reply k8s/tanka/environments/default/main.jsonnet.example:27]: Later, hence the TODO.

File: k8s/tanka/environments/default/tsbridge.jsonnet [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+                          },
+                        },
+                        {
+                          name: 'service-account-credentials-volume',
+                          secret: {
+                            secretName: 'google-api-credentials',
+                            items: [
+                              {
+                                key: 'json_key',
+                                path: 'google_key.json',
```
[the remark this answers is not in the document]
Comment (Temikus) [reply k8s/tanka/environments/default/tsbridge.jsonnet:95]: Not in this case (when we're mounting a secret) - the path itself is specified in the volume mount.

File: k8s/yaml/README.md (hunk 7 above)
Comment (Temikus) [reply k8s/yaml/README.md:11]: Done.

**Review by rjerrems — APPROVED** [review]
LGTM

File: k8s/yaml/deployment.yaml [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+      labels:
+        app: ts-bridge
+    spec:
+      securityContext:
+        runAsUser: 1000
+        runAsGroup: 1000
+        fsGroup: 1000
+      containers:
+        - image: REPLACE_ME
+          imagePullPolicy: Always
```
Comment (dparrish) [inline k8s/yaml/deployment.yaml:23]: This isn't ideal. The image tag should be unique so shouldn't need re-pulling. If you use `:latest` as the tag, k8s will always pull anyway. Removing this line would be saner IMHO.

File: k8s/yaml/deployment.yaml [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+            - --enable-status-page
+            - --stats-sd-project=REPLACE_ME
+            - --update-parallelism=4
+            - --sync-period=10s
+          ports:
+            - containerPort: 8080
+              name: ts-bridge
+          resources:
+            limits:
+              memory: "1Gi"
```
Comment (dparrish) [inline k8s/yaml/deployment.yaml:40]: Have you tested this to use between 1/2 and 1Gb RAM?

Interesting to note the k8s metric server won't expose correct usage for a job unless it has memory AND cpu resource constraints.

File: k8s/yaml/deployment.yaml [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+            requests:
+              memory: "500Mi"
+          livenessProbe:
```
Comment (dparrish) [inline k8s/yaml/deployment.yaml:43]: A readinessProbe will be good as well so the rollout can be checked. Also I believe readinessProbe is required for GCLB to balance to it if that's ever required.

File: k8s/yaml/deployment.yaml (hunk 12 above)
Comment (Temikus) [reply k8s/yaml/deployment.yaml:23]: That was a leftover from debugging, removed.

File: k8s/yaml/deployment.yaml (hunk 13 above)
Comment (Temikus) [reply k8s/yaml/deployment.yaml:40]: I'll remove the constraint for now, pending https://github.com/google/ts-bridge/issues/52

**Pull request merged**